src/main/java/org/sonarlint/intellij/issue/vulnerabilities/TaintVulnerabilityMatcher.kt

```diff
@@ -40,9 +40,16 @@ class TaintVulnerabilityMatcher(private val project: Project) {
   private fun matchLocation(location: ServerIssueLocation): Location {
     for (contentRoot in ProjectRootManager.getInstance(project)
       .contentRoots) {
-      val matchedFile = contentRoot.findFileByRelativePath(location.filePath!!)
-      if (matchedFile != null) {
-        return matchTextRange(matchedFile, location)
+      if (contentRoot.isDirectory()) {
+        val matchedFile = contentRoot.findFileByRelativePath(location.filePath!!)
```

Comment:
I suppose here `!!` is safe because we can be sure a taint will always have a file path

Reply:
It was already used few lines before, so I did not question that ;)